include/power/power_state.h

```diff
@@ -13,18 +13,153 @@
 extern "C" {
 #endif
 
-#define PM_STATE_RUNTIME_ACTIVE  BIT(0)
-#define PM_STATE_RUNTIME_IDLE    BIT(1)
-#define PM_STATE_SUSPEND_TO_IDLE BIT(2)
-#define PM_STATE_STANDBY         BIT(3)
-#define PM_STATE_SUSPEND_TO_RAM  BIT(4)
-#define PM_STATE_SUSPEND_TO_DISK BIT(5)
-#define PM_STATE_SOFT_OFF        BIT(6)
+/**
+ * @brief Runtime active state
+ *
+ * During runtime active state, the system is awake and running.
+ * In simple terms, the system is in a full running state.
+ */
+#define PM_STATE_RUNTIME_ACTIVE        BIT(0)
+
+/**
+ * @brief Runtime idle state
+ *
+ * Runtime idle is a system sleep state in which all of the cores
+ * enter deepest possible idle state and wait for interrupts, no
+ * requirements for the devices, leaving them at the states where
+ * they are.
+ */
+#define PM_STATE_RUNTIME_IDLE          BIT(1)
+
+/**
+ * @brief Suspend to idle state
+ *
+ * The system goes through a normal platform suspend where it puts
+ * all of the cores in deepest possible idle state and puts peripherals
+ * into low-power states. No operating state is lost (the cpu core
+ * retains power, does not lose execution context), so the system can
+ * go back to where it left off easily enough.
+ */
+#define PM_STATE_SUSPEND_TO_IDLE       BIT(2)
+
+/**
+ * @brief Standby state
+ *
+ * In addition to putting peripherals into low-power states, which is
+ * done for suspend to idle too, all non-boot CPUs are powered off.
+ * It should allow more energy to be saved relative to suspend to
+ * idle, but the resume latency will generally be greater than for that
+ * state. But it should be the same state with suspend to idle state on
+ * uniprocesser system.
+ */
+#define PM_STATE_STANDBY               BIT(3)
 
-#define PM_STATE_BIT_MASK        0x7F
+/**
+ * @brief Suspend to ram state
+ *
+ * This state offers significant energy savings by powering off as much
+ * of the system as possible, where memory should be placed into the
+ * self-refresh mode to retain its contents. The state of devices and
+ * CPUs is saved and held in memory, and it may require some boot-
```

Comment:
How is the state of the CPU saved?  Is this something that is supported by multiple platforms?

Generally, which of these modes are expected to be ones where the application will restart at the kernel boot and proceed through to main (unless something system-specific intervenes and allows it to jump back to a particular point in the code in an existing thread).

Reply:
multiple platforms can do this.